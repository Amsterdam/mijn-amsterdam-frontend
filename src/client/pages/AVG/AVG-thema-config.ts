import { generatePath } from 'react-router-dom';

import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';
import { withOmitDisplayPropsForSmallScreens } from '../../components/Table/helpers';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';
import { MAX_TABLE_ROWS_ON_THEMA_PAGINA } from '../../config/app';
import { TrackingConfig } from '../../config/routes';

const MAX_TABLE_ROWS_ON_THEMA_PAGINA_LOPEND = 5;

const displayPropsAanvragenBase: DisplayProps<
  WithDetailLinkComponent<AVGRequestFrontend>
> = {
  detailLinkComponent: 'Nummer',
  ontvangstDatumFormatted: 'Ontvangen op',
  themas: 'Onderwerp(en)',
};

const displayPropsAanvragen = withOmitDisplayPropsForSmallScreens(
  displayPropsAanvragenBase,
  ['themas']
);

const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
} as const;

export type ListPageParamKey = keyof typeof listPageParamKind;
export type ListPageParamKind = (typeof listPageParamKind)[ListPageParamKey];

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
} as const;

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_LOPEND,
    listPageRoute: generatePath(AppRoutes['AVG/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA,
    listPageRoute: generatePath(AppRoutes['AVG/LIST'], {
      kind: listPageParamKind.completed,
    }),
    ...tableConfigBase,
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
] as const;

export function getAVGListPageDocumentTitle(themaTitle: string) {
  return <T extends Record<string, string>>(
    config: TrackingConfig,
    params: T | null
  ) => {
    const kind = params?.kind as ListPageParamKind;
    return kind in tableConfig
      ? `${tableConfig[kind].title} | ${themaTitle}`
      : themaTitle;
  };
}
