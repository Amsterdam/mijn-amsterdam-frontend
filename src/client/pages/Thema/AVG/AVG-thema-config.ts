import { generatePath } from 'react-router';

import { AVGRequestFrontend } from '../../../../server/services/avg/types.ts';
import { dateSort } from '../../../../universal/helpers/date.ts';
import { capitalizeFirstLetter } from '../../../../universal/helpers/text.ts';
import { LinkProps } from '../../../../universal/types/App.types.ts';
import { type DisplayProps } from '../../../components/Table/TableV2.types.ts';
import {
  MAX_TABLE_ROWS_ON_THEMA_PAGINA,
  MAX_TABLE_ROWS_ON_THEMA_PAGINA_LOPEND,
} from '../../../config/app.ts';
import { type ThemaRoutesConfig } from '../../../config/thema-types.ts';

const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
} as const;

export const featureToggle = {
  avgActive: true,
};

export const themaId = 'AVG' as const;
export const themaTitle = 'AVG persoonsgegevens';

export const routeConfig = {
  detailPage: {
    path: '/avg/verzoek/:id',
    trackingUrl: '/avg/verzoek',
    documentTitle: `Avg verzoek | ${themaTitle}`,
  },
  listPage: {
    path: '/avg/lijst/:kind/:page?',
    documentTitle(params) {
      const kind = params?.kind as ListPageParamKind;
      return `${capitalizeFirstLetter(kind === 'lopende-aanvragen' ? 'Lopende' : 'Afgehanelde')} ${themaTitle} verzoeken | overzicht`;
    },
  },
  themaPage: {
    path: '/avg',
    documentTitle: `${themaTitle} verzoeken | overzicht`,
  },
} as const satisfies ThemaRoutesConfig;

const displayPropsAanvragen: DisplayProps<AVGRequestFrontend> = {
  props: {
    detailLinkComponent: 'Nummer',
    ontvangstDatumFormatted: 'Ontvangen op',
    themas: 'Onderwerp(en)',
  },
  colWidths: {
    large: ['15%', '35%', '50%'],
    small: ['20%', '80%', '0'],
  },
};

export type ListPageParamKey = keyof typeof listPageParamKind;
export type ListPageParamKind = (typeof listPageParamKind)[ListPageParamKey];

const tableConfigBase = {
  sort: dateSort('registratieDatum', 'desc'),
  displayProps: displayPropsAanvragen,
} as const;

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_LOPEND,
    listPageRoute: generatePath(routeConfig.listPage.path, {
      kind: listPageParamKind.inProgress,
      page: null,
    }),
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA,
    listPageRoute: generatePath(routeConfig.listPage.path, {
      kind: listPageParamKind.completed,
      page: null,
    }),
    ...tableConfigBase,
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
] as const;
