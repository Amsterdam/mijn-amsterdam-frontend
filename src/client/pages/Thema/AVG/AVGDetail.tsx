import { AVGRequestFrontend } from '../../../../server/services/avg/types';
import { Datalist } from '../../../components/Datalist/Datalist';
import { PageContentCell } from '../../../components/Page/Page';
import { useAVGDetailPage } from './useAVGDetailPage.hook';
import ThemaDetailPagina from '../../../components/Thema/ThemaDetailPagina';

function getVerzoekRows(verzoek: AVGRequestFrontend) {
  return [
    { label: 'Nummer', content: verzoek.id },
    { label: 'Type verzoek', content: verzoek.type },
    { label: 'Onderwerp(en)', content: verzoek.themas },
    { label: 'Toelichting', content: verzoek.toelichting },
  ].filter((row) => row.content);
}

function AVGDetailContent({ verzoek }: { verzoek: AVGRequestFrontend }) {
  const rows = getVerzoekRows(verzoek);

  return (
    <PageContentCell>
      <Datalist rows={rows} />
    </PageContentCell>
  );
}

function AVGDetail() {
  const { verzoek, isLoading, isError, breadcrumbs, title } =
    useAVGDetailPage();
  return (
    <ThemaDetailPagina
      title={title}
      zaak={verzoek}
      isError={isError}
      isLoading={isLoading}
      pageContentMain={!!verzoek && <AVGDetailContent verzoek={verzoek} />}
      breadcrumbs={breadcrumbs}
    />
  );
}

export { AVGDetail };
