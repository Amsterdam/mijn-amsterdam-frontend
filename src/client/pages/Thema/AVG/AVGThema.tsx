import { Paragraph } from '@amsterdam/design-system-react';

import { useAVGData } from './useAVGData.hook';
import { AVGRequestFrontend } from '../../../../server/services/avg/types';
import { PageContentCell } from '../../../components/Page/Page';
import ThemaPagina from '../../../components/Thema/ThemaPagina';
import ThemaPaginaTable from '../../../components/Thema/ThemaPaginaTable';
import { useHTMLDocumentTitle } from '../../../hooks/useHTMLDocumentTitle';

const pageContentTop = (
  <PageContentCell spanWide={8}>
    <Paragraph>Hieronder ziet u een overzicht van uw AVG verzoeken.</Paragraph>
  </PageContentCell>
);

export function AVGThema() {
  const {
    tableConfig,
    avgVerzoeken,
    isLoading,
    isError,
    linkListItems,
    title,
    routeConfig,
  } = useAVGData();
  useHTMLDocumentTitle(routeConfig.themaPage);

  const tables = Object.entries(tableConfig).map(
    ([
      kind,
      { title, displayProps, filter, sort, maxItems, listPageRoute },
    ]) => {
      return (
        <ThemaPaginaTable<AVGRequestFrontend>
          key={kind}
          title={title}
          listPageRoute={listPageRoute}
          zaken={avgVerzoeken.filter(filter).sort(sort)}
          displayProps={displayProps}
          maxItems={maxItems}
        />
      );
    }
  );

  return (
    <ThemaPagina
      title={title}
      isError={isError}
      isLoading={isLoading}
      pageContentTop={pageContentTop}
      pageContentMain={tables}
      linkListItems={linkListItems}
    />
  );
}
