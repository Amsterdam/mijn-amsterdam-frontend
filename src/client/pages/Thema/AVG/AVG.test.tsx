import { render } from '@testing-library/react';
import { generatePath } from 'react-router';
import { MutableSnapshot } from 'recoil';

import { routeConfig } from './AVG-thema-config';
import { testState } from './AVGDetail.test';
import { AVGThema } from './AVGThema';
import { AVGResponse } from '../../../../server/services/avg/types';
import { AppState } from '../../../../universal/types';
import { appStateAtom } from '../../../hooks/useAppState';
import MockApp from '../../MockApp';

function initializeState(testState: AppState) {
  return (snapshot: MutableSnapshot) => snapshot.set(appStateAtom, testState);
}

function setupTestComponent(testState: AppState) {
  const routeEntry = generatePath(routeConfig.themaPage.path);

  return function Component() {
    return (
      <MockApp
        routeEntry={routeEntry}
        routePath={routeEntry}
        component={AVGThema}
        initializeState={initializeState(testState)}
      />
    );
  };
}

describe('AVG thema pagina', () => {
  it('Matches the Full Page snapshot when there are requests', () => {
    const Component = setupTestComponent(testState);
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Matches the Full Page snapshot when there are no requests', () => {
    const themaContent: AVGResponse = {
      verzoeken: [],
      aantal: 0,
    };
    const Component = setupTestComponent({
      AVG: {
        status: 'OK',
        content: themaContent,
      },
    } as unknown as AppState);

    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Matches the Full Page snapshot when API gives no result', () => {
    const Component = setupTestComponent({
      AVG: {
        status: 'ERROR',
        content: null,
      },
    } as unknown as AppState);
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
