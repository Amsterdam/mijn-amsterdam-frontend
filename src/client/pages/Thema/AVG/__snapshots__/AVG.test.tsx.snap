// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AVG thema pagina > Matches the Full Page snapshot when API gives no result 1`] = `
<DocumentFragment>
  <div
    class="ams-grid"
  >
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide _PageHeadingWrap_c5a3f1"
    >
      <div
        class="_PageHeadingInner_c5a3f1"
        id="skip-to-id-AppContent"
      >
        <nav
          class="ams-breadcrumb"
        >
          <ol
            class="ams-breadcrumb__list"
          >
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/avg"
                tabindex="-1"
              >
                <span
                  class="ams-visually-hidden"
                >
                  AVG persoonsgegevens
                </span>
              </a>
            </li>
          </ol>
        </nav>
        <h3
          class="ams-heading ams-heading--level-1 _PageHeading_c5a3f1"
        >
          AVG persoonsgegevens
        </h3>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-8-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <p
        class="ams-paragraph"
      >
        Hieronder ziet u een overzicht van uw AVG verzoeken.
      </p>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <ul
        class="ams-link-list"
      >
        <li>
          <a
            class="ams-link-list__link"
            href="https://www.amsterdam.nl/privacy/loket/"
            rel="noreferrer"
          >
            <span
              class="ams-icon ams-icon--size-5"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9.757 32-2.9-2.91L19.937 16 6.857 2.91 9.757 0l16 16z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            Loket persoonsgegevens gemeente Amsterdam
          </a>
        </li>
      </ul>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <section
        class="ams-alert ams-alert--error"
      >
        <div
          class="ams-alert__icon"
        >
          <span
            class="ams-icon ams-icon--size-4"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              viewBox="0 0 36 36"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M33.63 30.9H2.37L18 3zm-24.44-4h17.62L18 11.19z"
              />
              <path
                d="M17 16.99h2v4.95h-2zm0 5.95h2v1.95h-2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="ams-alert__content"
        >
          <h2
            class="ams-heading ams-heading--level-4"
          >
            Foutmelding
          </h2>
          <p
            class="ams-paragraph"
          >
            We kunnen op dit moment niet alle gegevens tonen.
          </p>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AVG thema pagina > Matches the Full Page snapshot when there are no requests 1`] = `
<DocumentFragment>
  <div
    class="ams-grid"
  >
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide _PageHeadingWrap_c5a3f1"
    >
      <div
        class="_PageHeadingInner_c5a3f1"
        id="skip-to-id-AppContent"
      >
        <nav
          class="ams-breadcrumb"
        >
          <ol
            class="ams-breadcrumb__list"
          >
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/avg"
                tabindex="-1"
              >
                <span
                  class="ams-visually-hidden"
                >
                  AVG persoonsgegevens
                </span>
              </a>
            </li>
          </ol>
        </nav>
        <h3
          class="ams-heading ams-heading--level-1 _PageHeading_c5a3f1"
        >
          AVG persoonsgegevens
        </h3>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-8-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <p
        class="ams-paragraph"
      >
        Hieronder ziet u een overzicht van uw AVG verzoeken.
      </p>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <ul
        class="ams-link-list"
      >
        <li>
          <a
            class="ams-link-list__link"
            href="https://www.amsterdam.nl/privacy/loket/"
            rel="noreferrer"
          >
            <span
              class="ams-icon ams-icon--size-5"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9.757 32-2.9-2.91L19.937 16 6.857 2.91 9.757 0l16 16z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            Loket persoonsgegevens gemeente Amsterdam
          </a>
        </li>
      </ul>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <h3
        class="ams-heading ams-heading--level-2"
      >
        Lopende aanvragen
      </h3>
      <div
        class="ams-table"
      >
        <table
          class="ams-table__table _TableV2_4c6825"
        >
          <tbody
            class="ams-table__body"
          />
        </table>
      </div>
      <p
        class="ams-paragraph"
      >
        U heeft (nog) geen lopende aanvragen
      </p>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <h3
        class="ams-heading ams-heading--level-2"
      >
        Afgehandelde aanvragen
      </h3>
      <div
        class="ams-table"
      >
        <table
          class="ams-table__table _TableV2_4c6825"
        >
          <tbody
            class="ams-table__body"
          />
        </table>
      </div>
      <p
        class="ams-paragraph"
      >
        U heeft (nog) geen afgehandelde aanvragen
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AVG thema pagina > Matches the Full Page snapshot when there are requests 1`] = `
<DocumentFragment>
  <div
    class="ams-grid"
  >
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide _PageHeadingWrap_c5a3f1"
    >
      <div
        class="_PageHeadingInner_c5a3f1"
        id="skip-to-id-AppContent"
      >
        <nav
          class="ams-breadcrumb"
        >
          <ol
            class="ams-breadcrumb__list"
          >
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/avg"
                tabindex="-1"
              >
                <span
                  class="ams-visually-hidden"
                >
                  AVG persoonsgegevens
                </span>
              </a>
            </li>
          </ol>
        </nav>
        <h3
          class="ams-heading ams-heading--level-1 _PageHeading_c5a3f1"
        >
          AVG persoonsgegevens
        </h3>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-8-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <p
        class="ams-paragraph"
      >
        Hieronder ziet u een overzicht van uw AVG verzoeken.
      </p>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <ul
        class="ams-link-list"
      >
        <li>
          <a
            class="ams-link-list__link"
            href="https://www.amsterdam.nl/privacy/loket/"
            rel="noreferrer"
          >
            <span
              class="ams-icon ams-icon--size-5"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                viewBox="0 0 32 32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9.757 32-2.9-2.91L19.937 16 6.857 2.91 9.757 0l16 16z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            Loket persoonsgegevens gemeente Amsterdam
          </a>
        </li>
      </ul>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <h3
        class="ams-heading ams-heading--level-2"
      >
        Lopende aanvragen
      </h3>
      <div
        class="ams-table"
      >
        <table
          class="ams-table__table _TableV2_4c6825"
        >
          <thead
            class="ams-table__header"
          >
            <tr
              class="ams-table__row"
            >
              <th
                class="ams-table__header-cell"
              >
                Nummer
              </th>
              <th
                class="ams-table__header-cell"
              >
                Ontvangen op
              </th>
              <th
                class="ams-table__header-cell"
              >
                Onderwerp(en)
              </th>
            </tr>
          </thead>
          <tbody
            class="ams-table__body"
          >
            <tr
              class="ams-table__row"
            >
              <td
                class="ams-table__cell"
              >
                <a
                  class="ams-link ams-link--standalone _MaRouterLink__no-underline-fat_aa758a"
                  href="/avg/verzoek/1"
                  title="Bekijk meer over avg verzoek met nummer 1"
                >
                  1
                </a>
              </td>
              <td
                class="ams-table__cell"
              >
                06 maart 2023
              </td>
              <td
                class="ams-table__cell"
              >
                Parkeren, Foo, Bar
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <h3
        class="ams-heading ams-heading--level-2"
      >
        Afgehandelde aanvragen
      </h3>
      <div
        class="ams-table"
      >
        <table
          class="ams-table__table _TableV2_4c6825"
        >
          <thead
            class="ams-table__header"
          >
            <tr
              class="ams-table__row"
            >
              <th
                class="ams-table__header-cell"
              >
                Nummer
              </th>
              <th
                class="ams-table__header-cell"
              >
                Ontvangen op
              </th>
              <th
                class="ams-table__header-cell"
              >
                Onderwerp(en)
              </th>
            </tr>
          </thead>
          <tbody
            class="ams-table__body"
          >
            <tr
              class="ams-table__row"
            >
              <td
                class="ams-table__cell"
              >
                <a
                  class="ams-link ams-link--standalone _MaRouterLink__no-underline-fat_aa758a"
                  href="/avg/verzoek/223"
                  title="Bekijk meer over avg verzoek met nummer 223"
                >
                  223
                </a>
              </td>
              <td
                class="ams-table__cell"
              >
                09 maart 2022
              </td>
              <td
                class="ams-table__cell"
              >
                Vergunningen, Enzo
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AVGDetail > No result > Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <div
    class="ams-grid"
  >
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide _PageHeadingWrap_c5a3f1"
    >
      <div
        class="_PageHeadingInner_c5a3f1"
        id="skip-to-id-AppContent"
      >
        <nav
          class="ams-breadcrumb"
        >
          <ol
            class="ams-breadcrumb__list"
          >
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/avg"
              >
                AVG persoonsgegevens
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/avg/verzoek/abc"
                tabindex="-1"
              >
                <span
                  class="ams-visually-hidden"
                >
                  AVG verzoek
                </span>
              </a>
            </li>
          </ol>
        </nav>
        <h3
          class="ams-heading ams-heading--level-1 _PageHeading_c5a3f1"
        >
          AVG verzoek
        </h3>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <section
        class="ams-alert ams-alert--error"
      >
        <div
          class="ams-alert__icon"
        >
          <span
            class="ams-icon ams-icon--size-4"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              viewBox="0 0 36 36"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M33.63 30.9H2.37L18 3zm-24.44-4h17.62L18 11.19z"
              />
              <path
                d="M17 16.99h2v4.95h-2zm0 5.95h2v1.95h-2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="ams-alert__content"
        >
          <h2
            class="ams-heading ams-heading--level-4"
          >
            Foutmelding
          </h2>
          <p
            class="ams-paragraph"
          >
            We kunnen op dit moment geen gegevens tonen.
          </p>
        </div>
      </section>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AVGDetail > With result > Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <div
    class="ams-grid"
  >
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide _PageHeadingWrap_c5a3f1"
    >
      <div
        class="_PageHeadingInner_c5a3f1"
        id="skip-to-id-AppContent"
      >
        <nav
          class="ams-breadcrumb"
        >
          <ol
            class="ams-breadcrumb__list"
          >
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/avg"
              >
                AVG persoonsgegevens
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/avg/verzoek/223"
                tabindex="-1"
              >
                <span
                  class="ams-visually-hidden"
                >
                  AVG verzoek
                </span>
              </a>
            </li>
          </ol>
        </nav>
        <h3
          class="ams-heading ams-heading--level-1 _PageHeading_c5a3f1"
        >
          AVG verzoek
        </h3>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <dl
        class="_Datalist_08bf39"
      >
        <dt
          class="_Datalist__title_08bf39"
        >
          Nummer
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          223
        </dd>
        <dt
          class="_Datalist__title_08bf39"
        >
          Type verzoek
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          Verwijderen gegevens
        </dd>
        <dt
          class="_Datalist__title_08bf39"
        >
          Onderwerp(en)
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          Vergunningen, Enzo
        </dd>
      </dl>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <div
        class="ams-grid _Steps_7a365b"
      >
        <h1
          class="ams-heading ams-heading--level-2 _StatusHeading_7a365b ams-mb--sm"
        >
          Status
        </h1>
        <div
          class="ams-grid__cell ams-grid__cell--span-10 ams-grid__cell--start-2"
        >
          <ol
            class="ams-ordered-list ams-ordered-list--no-markers _StepsList_7a365b"
          >
            <li
              class="ams-ordered-list__item _Step_7a365b _Step--checked_7a365b"
            >
              <h4
                class="ams-heading ams-heading--level-4 _StepStatus_7a365b"
              >
                <span
                  aria-label="Status Afgerond"
                  class="_StepStatusIndication_7a365b"
                >
                  <span
                    aria-label="Status Afgerond"
                    class="ams-icon ams-icon--size-6 _StatusIndicationCheckmark_7a365b"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      viewBox="0 0 32 32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12.216 27.016 0 14.168l2.91-2.77 9.346 9.837L29.129 4 32 6.8z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </span>
                Ontvangen
              </h4>
              <time
                class="_StepStatusDate_7a365b"
                datetime="2023-03-18T00:00:00.000Z"
              >
                18 maart 2023
              </time>
            </li>
            <li
              class="ams-ordered-list__item _Step_7a365b _Step--checked_7a365b"
            >
              <h4
                class="ams-heading ams-heading--level-4 _StepStatus_7a365b"
              >
                <span
                  aria-label="Status Afgerond"
                  class="_StepStatusIndication_7a365b"
                >
                  <span
                    aria-label="Status Afgerond"
                    class="ams-icon ams-icon--size-6 _StatusIndicationCheckmark_7a365b"
                  >
                    <svg
                      aria-hidden="true"
                      focusable="false"
                      viewBox="0 0 32 32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12.216 27.016 0 14.168l2.91-2.77 9.346 9.837L29.129 4 32 6.8z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </span>
                In behandeling
              </h4>
              <time
                class="_StepStatusDate_7a365b"
                datetime="2023-05-30T00:00:00.000Z"
              >
                30 mei 2023
              </time>
            </li>
            <li
              class="ams-ordered-list__item _Step_7a365b _Step--active_7a365b"
            >
              <h4
                class="ams-heading ams-heading--level-4 _StepStatus_7a365b"
              >
                <span
                  aria-label="Huidige status"
                  class="_StepStatusIndication_7a365b"
                />
                Extra informatie nodig
              </h4>
              <time
                class="_StepStatusDate_7a365b"
                datetime="2023-06-03T00:00:00.000Z"
              >
                03 juni 2023
              </time>
              <div>
                <div
                  class=""
                >
                  Wij hebben meer informatie nodig om uw verzoek in behandeling te nemen. U krijgt een brief waarin staat welke informatie wij nodig hebben.
                </div>
              </div>
            </li>
            <li
              class="ams-ordered-list__item _Step_7a365b"
            >
              <h4
                class="ams-heading ams-heading--level-4 _StepStatus_7a365b"
              >
                <span
                  aria-label="Toekomstige status"
                  class="_StepStatusIndication_7a365b"
                />
                Afgehandeld
              </h4>
              <time
                class="_StepStatusDate_7a365b"
                datetime=""
              />
            </li>
          </ol>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
